src/main/io/displayport_msp.c

```diff
@@ -168,6 +168,32 @@ static uint32_t txBytesFree(const displayPort_t *displayPort)
     return mspSerialTxBytesFree();
 }
 
+#ifdef USE_MSP_DISPLAYPORT_FONT
+static bool writeFontCharacter(displayPort_t *displayPort, uint16_t addr, const osdCharacter_t *chr)
+{
+    int res;
+    uint8_t buf[OSD_CHAR_BYTES + 4];
+
+    if (!chr) {
+        return false;
+    }
+    
+    buf[0] = MSP_DP_CHAR_WRITE;
+    buf[1] = addr & 0xff;
+    buf[2] = (addr >> 8) & 0xff;
+    buf[3] = 0;
+
+    memcpy(&buf[4], (char *)chr, OSD_CHAR_BYTES);
+
+    res = output(displayPort, MSP_DISPLAYPORT, buf, OSD_CHAR_BYTES + 4);
```

Comment:
```suggestion
    uint8_t *p = buf;
    *p++ = MSP_DP_CHAR_WRITE;
    *p++ = addr & 0xff;
    *p++ = (addr >> 8) & 0xff;
    *p++ = 0;

    memcpy(p, chr->data, OSD_CHAR_BYTES);
    p += OSD_CHAR_BYTES;
    int res = output(displayPort, MSP_DISPLAYPORT, buf, p - buf);
```

Reply:
I prefer the original code, although the array indices would be better defined as constants.

Reply:
@SteveCEvans : PACKED struct is IMO superior of byte array indices. Building packet in parts using pointer is IMO a bit more resilient to refactoring and common errors (`struct`+sizeof will provide similar protection)

Reply:
i will refactor it to a struct :)